src/core/lib/surface/completion_queue.cc

```diff
@@ -780,10 +780,11 @@ typedef struct {
   bool first_loop;
 } cq_is_finished_arg;
 
-class ExecCtxNext : public grpc_core::ExecCtx {
+class ExecCtxNext final : public grpc_core::ExecCtx {
```

Comment:
why does this class (and `ExecCtxPluck`) need be final? Do these changes still belong to this PR?

Reply:
No, we can take out the `final` part for now. That should be a different PR.

Reply:
I removed the changes that were present in `completion_queue.cc` as they weren't needed for this PR.